cNVMe/Namespace.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+###########################################################################################
+// cNVMe - An Open Source NVMe Device Simulation - MIT License
+// Copyright 2018 - Intel Corporation
+
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+// PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT
+// OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+############################################################################################
+Namespace.cpp - An implementation file for a cNVMe Namespace
+*/
+
+#define DEFAULT_NUMBER_OF_LBA_FORMAT 2; // 0-based!
+
+#include "Constants.h"
+#include "Namespace.h"
+
+namespace cnvme
+{
+	namespace ns
+	{
+		Namespace::Namespace()
+		{
+			memset(&this->IdentifyNamespace, 0, sizeof(this->IdentifyNamespace));
+		}
+
+		Namespace::Namespace(size_t SizeInBytes) : Namespace()
+		{
+			Media = Payload(SizeInBytes);
+		}
+
+		Namespace::~Namespace()
+		{
+			// nothing to do yet.
+		}
+
+		identify::structures::IDENTIFY_NAMESPACE& Namespace::getIdentifyNamespaceStructure()
+		{
+			this->IdentifyNamespace.NLBAF = DEFAULT_NUMBER_OF_LBA_FORMAT; // support 512/4096/8192 byte sectors
+			this->IdentifyNamespace.LBAF[0].LBADS = 9; // 2^9 = 512
+			this->IdentifyNamespace.LBAF[1].LBADS = 12; // 2^12 = 4096
+			this->IdentifyNamespace.LBAF[2].LBADS = 13; // 2^13 = 8192
+
+			auto currentLbaFormat = this->IdentifyNamespace.LBAF[this->IdentifyNamespace.FLBAS.CurrentLBAFormat];
+			ASSERT_IF(currentLbaFormat.LBADS < 9, "Minimum posssible selected LBADS should be 9. 2^9 = 512.");
```

Comment:
How is it possible to hit this assert? Is this just for development?

Reply:
It really isn't just for dev/if something goes very wrong.